fix confirm-quit=downloads with finished downloads

Kingdread opened the pull request:
Fixes #846

`.rowCount()` returns all downloads, even the finished ones that have not
yet been removed from the list. For confirming the quit event, we should
only consider downloads that are still running.

Kingdread commented:
Also, something I noticed, in [downloads.py, lines 900-904](https://github.com/The-Compiler/qutebrowser/blob/master/qutebrowser/browser/downloads.py#L900-L904) it says

```
        idx = len(self.downloads) + 1
        download.index = idx
        self.beginInsertRows(QModelIndex(), idx, idx)
        self.downloads.append(download)
        self.endInsertRows()
```

`len(self.downloads)` is already the new index for the appended item (e.g. if the list is empty, the new item will have index 0, and so on), so `idx` is the "humand readable" index (1 based), but if I understand [the Qt docs right](http://doc.qt.io/qt-5/qabstractitemmodel.html#beginInsertRows), we should pass the new index to `beginInsertRows`, which should be `idx - 1` (or just `len(self.downloads)`).

Not sure if that would change anything (I doubt it), I just noticed it (and wasn't 100% sure, so I didn't want to change it yet)

The-Compiler commented on qutebrowser/browser/downloads.py:
```diff
@@ -1238,3 +1238,11 @@ def rowCount(self, parent=QModelIndex()):
+        Return:
+            The number of unfinished downloads.
+        """
+        return len([1 for download in self.downloads if not download.done])
```
The usual way this kind of thing is done is `sum(1 for download in self.downloads if not download.done)` - it will get you the same, but without building a list.

The-Compiler commented:
I think you're right with what you mention in your comment. I'm somewhat surprised this didn't cause any trouble yet :wink:

I think it'd make sense to keep the human-readable index for `downloads.index`, but for the indices passed to `beginInsertRows` it seems wrong indeed.

Kingdread commented:
I've updated the PR with the `sum` and `idx` changes.

I've noted that I accidentally based this branch off 2918c5c (downloads: close fileobject in DownloadItem._die) from the other PR (#1473), which is merged. GitHub didn't complain, but I rebased onto master anyway (just for the protocol :wink:)

The-Compiler commented:
LGTM, but I'll go to bed now and merge tomorrow :wink: - thanks!

The-Compiler commented:
Thanks! I also added a test in f7dc9b5.
